ui/pages/confirmations/hooks/useTrackERC20WithoutDecimalInformation.ts

```diff
@@ -0,0 +1,58 @@
+import { useSelector } from 'react-redux';
+import { useContext, useEffect } from 'react';
+import { Hex } from '@metamask/utils';
+
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventLocation,
+  MetaMetricsEventName,
+  MetaMetricsEventUiCustomization,
+} from '../../../../shared/constants/metametrics';
+import { TokenStandard } from '../../../../shared/constants/transaction';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { getCurrentChainId } from '../../../selectors';
+import { parseTokenDetailDecimals, TokenDetailsERC20 } from '../utils/token';
+
+/**
+ * Track event that number of decimals in ERC20 is not obtained
+ *
+ * @param tokenAddress
+ * @param tokenDetails
+ * @param metricLocation
+ */
+const useTrackERC20WithoutDecimalInformation = (
+  tokenAddress: Hex | string | undefined,
+  tokenDetails?: TokenDetailsERC20,
+  metricLocation = MetaMetricsEventLocation.SignatureConfirmation,
+) => {
+  const trackEvent = useContext(MetaMetricsContext);
+  const chainId = useSelector(getCurrentChainId);
+
+  useEffect(() => {
+    if (chainId === undefined || tokenDetails === undefined) {
+      return;
+    }
+    const { decimals, standard } = tokenDetails || {};
+    if (standard === TokenStandard.ERC20) {
```

Comment:
Minor, rather than nesting the majority of the block, could we early return if standard isn't ERC-20?

Reply:
In either case we need to return `details` object. Thus this looks more simplified.